### DISCUSSION TITLE
Trying not to deploy an api on the client machine.

### DISCUSSION THREAD

**Discussion by kcabral817** [MarimerLLC/csla#1718] [Questions] [answered]
I have an api (smartystreets) that I use to verify and format addresses.  I have set up a proxy in an attempt to NOT deploy the smartyStreets api on the client machine.  This idea was a dismal failure.  Because I have a business rule that invokes the proxy that invokes the class that has the SmartyStreets api in it, I get an error when using the UI with "file not found, you must include SmartyStreets" yada yada, yada.  I thought a proxy would solve the problem--but not.  Does anyone have any idea how I can NOT deploy the api to client machine but still get the functionality to run on the server?  My AddressVerificationManager is a commandBase class that has the Using SmartyStreets in it.  I don't call this class directly from the rules nor the front end.

**Comment by rockfordlhotka** [comment, marked as answer]
I'd like to confirm some things:

1. You are using a remote data portal with app server
1. smartystreets is only installed on the app server
1. You don't reference smartystreets in
   1. the client-side app
   1. the business layer
1. Your command object's `Execute` operation method (running on the app server) calls some assembly that is only deployed on the app server, and that assembly talks to the smartystreets API

**Reply by kcabral817** [reply]
1.  Yes.  Invoke:  ((Csla.Core.ICheckRules)Model.BankAddressList[e.Record.Index].Address).CheckRules(); on each address line to validate in the UI.
2.  Yes.  Not on the client.
3.  i.  No. 
     ii.  Yes.  
protected override void AddBusinessRules()
        {
            BusinessRules.AddRule(new AddressValidationRule(props) { Priority = 1});
         }
4.  It's not in its own assembly. It is in its own class in the same project as the AddressValidationRule class that does a DataPortal.Create & Execute to AddressVerificationManager.  In AVM it has the using statement for SmartyStreets and is the CommandBase<AddressVerificationManager>.  

        AddressValidationRule : ObjectRule
        private async Task<AddressInfoPersist> VerifyAddress(string addressLine)
        {
            var cmd = DataPortal.Create<AddressVerificationManager>(addressLine);

            cmd = await DataPortal.ExecuteAsync(cmd);

            return cmd.ConfiguredAddress;
        }

        AddressVerificationManager : CommandBase<AddressVerificationManager>
 
        using SmartyStreets;
        using SmartyStreets.USStreetApi;
